src/xref2/link.ml

```diff
@@ -15,13 +15,24 @@ let synopsis_from_comment (docs : Component.CComment.docs) =
       Comment.synopsis [ e ]
   | _ -> None
 
-let synopsis_of_module env (m : Component.Module.t) =
+let synopsis_of_module env id (m : Component.Module.t) =
+  let open Utils.ResultMonad in
   match synopsis_from_comment m.doc with
   | Some _ as s -> s
   | None -> (
       (* If there is no doc, look at the expansion. *)
-      match Tools.signature_of_module env m with
-      | Ok sg -> synopsis_from_comment (Component.extract_signature_doc sg)
+      match
+        Tools.expansion_of_module env m >>= Expand_tools.handle_expansion env id
```

Comment:
This could use the "handle_expansion_sig" to avoid computing the arguments.

Reply:
Ironically this bit that is actually using `Expand_tools.handle_expansion` doesn't need it, I think.